linux/driver/osp-iio.c

```diff
@@ -0,0 +1,1203 @@
+	int usebuffer;
+	int useevent;
+} and_sensor[NUM_ANDROID_SENSOR_TYPE] = {
+	[SENSOR_GYROSCOPE] = {
+		.name = "osp_gyro",
+		.trigname = "osp_gyro",
+		.channels = gyro_channels,
+		.num_channels = ARRAY_SIZE(gyro_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_GYROSCOPE_UNCALIBRATED] = {
+		.name = "osp_uncal_gyro",
+		.trigname = "osp_uncal_gyro",
+		.channels = gyro_channels,
+		.num_channels = ARRAY_SIZE(gyro_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_ACCELEROMETER] = {
+		.name = "osp_accel",
+		.trigname = "osp_accel",
+		.channels = accel_channels,
+		.num_channels = ARRAY_SIZE(accel_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_GRAVITY] = {
+		.name = "osp_gravity",
+		.trigname = "osp_gravity",
+		.channels = accel_channels,
+		.num_channels = ARRAY_SIZE(accel_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_MAGNETIC_FIELD] = {
+		.name = "osp_mag",
+		.trigname = "osp_mag",
+		.channels = mag_channels,
+		.num_channels = ARRAY_SIZE(mag_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_MAGNETIC_FIELD_UNCALIBRATED] = {
+		.name = "osp_uncal_mag",
+		.trigname = "osp_uncal_mag",
+		.channels = mag_channels,
+		.num_channels = ARRAY_SIZE(mag_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_ORIENTATION] = {
+		.name = "osp_compass_orientation",
+		.trigname = "osp_compass_orientation",
+		.channels = orient_channels,
+		.num_channels = ARRAY_SIZE(mag_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_LINEAR_ACCELERATION] = {
+		.name = "osp_linear_acceleration",
+		.trigname = "osp_linear_acceleration",
+		.channels = linacc_channels,
+		.num_channels = ARRAY_SIZE(linacc_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_ROTATION_VECTOR] = {
+		.name = "osp_rotation_vector",
+		.trigname = "osp_rotation_vector",
+		.channels = rotvec_channels,
+		.num_channels = ARRAY_SIZE(rotvec_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_GAME_ROTATION_VECTOR] = {
+		.name = "osp_game_rotation_vector",
+		.trigname = "osp_game_rotation_vector",
+		.channels = rotvec_channels,
+		.num_channels = ARRAY_SIZE(rotvec_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_GEOMAGNETIC_ROTATION_VECTOR] = {
+		.name = "osp_geo_rotation_vector",
+		.trigname = "osp_geo_rotation_vector",
+		.channels = rotvec_channels,
+		.num_channels = ARRAY_SIZE(rotvec_channels),
+		.info = &osp_sensor_info,
+		.usebuffer = 1,
+		.useevent = 0,
+	},
+	[SENSOR_SIGNIFICANT_MOTION] = {
+		.name = "osp_sigmot",
+		.trigname = "osp_sigmot",
+		.channels = NULL,
+		.num_channels = 0,
+		.info = &osp_sensor_info,
+		.usebuffer = 0,
+		.useevent = 1,
```

Comment:
Instead of setting number channels 1, is it possible to assign channels for device which can report events and we can specify channel info -> type of channels, modifier, direction and event_mask to describe what kind of events this channel can report.